lib/ZnapZend/ZFS.pm

```diff
@@ -553,7 +557,75 @@ sub sendRecvSnapshots {
     return 1;
 }
 
+sub filterPropertyNames {
+    # This routine is a helper for getXproperties to allow easy passing of
+    # specific properties we are interested to `zfs get`, to optimize some
+    # code around that. As an input it can get either:
+    # * string that would be passed through as is (after some sanitization);
+    # * array of strings that would be concatenated into a comma-separated
+    #   string with our name-space propertyPrefix (e.g. "org.znapzend:...")
+    #   prepended if needed;
+    # * undef for defaulting to 'all'
+    # Returns a string safe to pass into `zfs get`
+    my $self = shift;
+    my $propnames = shift;
+
+    my $propertyPrefix = $self->propertyPrefix;
+
+    if (defined($propnames) && $propnames) {
+        if (ref($propnames) eq 'ARRAY') {
+            if (scalar(@$propnames) > 0) {
+                my $propstring = '';
+                foreach my $propname (@$propnames) {
+                    next if !defined($propname);
+                    chomp $propname;
+
+                    if ($propname eq 'all') {
+                        warn "=== getSnapshotProperties(): got an 'all' propname in array, any filtering will be moot";
```

Comment:
using 'warn' can be problematic as it does not go to the log ...

Reply:
Note to self: I'll revise the use of `print STDERR` too - there was a lot of such copypasta :(

Reply:
Uff, that was an unexpected long adventure :)